jf_agent/git/gitlab_adapter.py

```diff
@@ -428,42 +432,52 @@ def _normalize_commit(
 def _get_normalized_pr_comments(
     merge_request, strip_text_content
 ) -> List[NormalizedPullRequestComment]:
-    try:
-        return [
-            NormalizedPullRequestComment(
-                user=_normalize_user(note.author),
-                body=sanitize_text(note.body, strip_text_content),
-                created_at=note.created_at,
-                system_generated=note.system,
+    if merge_request.note_list:
+        try:
+            return [
+                NormalizedPullRequestComment(
+                    user=_normalize_user(note.author),
+                    body=sanitize_text(note.body, strip_text_content),
+                    created_at=note.created_at,
+                    system_generated=note.system,
+                )
+                for note in merge_request.note_list
+            ]
+        except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
+            log_and_print_request_error(
+                e,
+                f'standardizing PR comments for merge_request {merge_request.id} -- '
+                f'handling it as if it has no comments',
             )
-            for note in merge_request.note_list
-        ]
-    except (requests.exceptions.RetryError, gitlab.exceptions.GitlabHttpError) as e:
-        log_and_print_request_error(
-            e,
-            f'standardizing PR comments for merge_request {merge_request.id} -- '
-            f'handling it as if it has no comments',
-        )
-        return []
+            return []
+    print(
+        f'\nNo comments found. Standardizing PR comments for merge_request ID: {merge_request.id} -- \n handling it as if it has no comments'
+    )
+    return []
```

Comment:
Looks like we can keep the old code here unchanged, with the exception of changing this line:
```
for note in merge_request.note_list
```

To this:
```
for note in merge_request.note_list or []
```

And that should be equivalent to the new code that you've written here.

That'd be somewhat more appealing since we'd just be changing less code (introducing less risk), and keeping the code a bit more brief.

Reply:
Agreed, I think that try/except was creating a TypeError with the v3 objects and rather that figuring out how to get them to conform to the expected data structure I just wrapped the block with another conditional check. I changed enough other stuff during that last round of testing that it may be worth giving this a shot so I'll run it now and update in a few.

Reply:
Following up, it worked with your recommended changes!